### DISCUSSION TITLE
[many ports] Dirty share folder

### DISCUSSION THREAD

**Discussion by Thomas1664** [microsoft/vcpkg#23619] [Ideas]
**Describe the bug**
Many ports leave docs/man/test files in `${CURRENT_PACKAGES_DIR}/share` folder.
For example:

```
wordnet:x64-linux:/share/doc/wordnet/Makefile.in
wordnet:x64-linux:/share/doc/wordnet/html/Makefile
wordnet:x64-linux:/share/doc/wordnet/html/Makefile.am
wordnet:x64-linux:/share/doc/wordnet/html/Makefile.in
wordnet:x64-linux:/share/doc/wordnet/html/binsrch.3WN.html
wordnet:x64-linux:/share/doc/wordnet/html/cntlist.5WN.html
wordnet:x64-linux:/share/doc/wordnet/html/grind.1WN.html
wordnet:x64-linux:/share/doc/wordnet/html/lexnames.5WN.html
wordnet:x64-linux:/share/doc/wordnet/html/morph.3WN.html
wordnet:x64-linux:/share/doc/wordnet/html/morphy.7WN.html
wordnet:x64-linux:/share/doc/wordnet/html/prologdb.5WN.html
wordnet:x64-linux:/share/doc/wordnet/html/senseidx.5WN.html
wordnet:x64-linux:/share/doc/wordnet/html/uniqbeg.7WN.html
wordnet:x64-linux:/share/doc/wordnet/html/wn.1WN.html
wordnet:x64-linux:/share/doc/wordnet/html/wnb.1WN.html
wordnet:x64-linux:/share/doc/wordnet/html/wndb.5WN.html
wordnet:x64-linux:/share/doc/wordnet/html/wngloss.7WN.html
wordnet:x64-linux:/share/doc/wordnet/html/wngroups.7WN.html
wordnet:x64-linux:/share/doc/wordnet/html/wninput.5WN.html
wordnet:x64-linux:/share/doc/wordnet/html/wnintro.1WN.html
wordnet:x64-linux:/share/doc/wordnet/html/wnintro.3WN.html
wordnet:x64-linux:/share/doc/wordnet/html/wnintro.5WN.html
wordnet:x64-linux:/share/doc/wordnet/html/wnintro.7WN.html
wordnet:x64-linux:/share/doc/wordnet/html/wnlicens.7WN.html
wordnet:x64-linux:/share/doc/wordnet/html/wnpkgs.7WN.html
wordnet:x64-linux:/share/doc/wordnet/html/wnsearch.3WN.html
wordnet:x64-linux:/share/doc/wordnet/html/wnstats.7WN.html
wordnet:x64-linux:/share/doc/wordnet/html/wnutil.3WN.html
wordnet:x64-linux:/share/doc/wordnet/man/Makefile
wordnet:x64-linux:/share/doc/wordnet/man/Makefile.am
wordnet:x64-linux:/share/doc/wordnet/man/Makefile.in
```

**To reproduce**
Grab file lists from any CI run and search for `.html` in `VCPKGDatabase.txt`. You will get many results for documentation files.
[VCPKGDatabase.txt](https://github.com/microsoft/vcpkg/files/8253802/VCPKGDatabase.txt)

**Solution**
- Delete those folders/ files from within `portfile.cmake`
- Remove `${CURRENT_PACKAGES_DIR}/share/${PORT}/docs` and similar directories by default
- Add a post build check that checks for `.html` or similar files

**Comment by dg0yt** [comment]
I would suggest the following priorities:
1. Create a PR check for doc, html, man directories in `/share`.
   (The wording of the suggestion should include "If you modified port ${PORT}, ...".)
   In the meantime, reviewers might check this manually.
2. Fix the undesired directories only when updating port versions, to limit impact on users.
3. Implement a post-build check (controlled by policy).

**Comment by Thomas1664** [comment]
> 1. Create a PR check for doc, html, man directories in `/share`.
>    (The wording of the suggestion should include "If you modified port ${PORT}, ...".)
>    In the meantime, reviewers might check this manually.

@dg0yt I agree with you, but I think it is a bad idea to rely on certain directory names that are forbidden because some port might have `man` or similar in their name or the documentation folder is nested somewhere. Furthermore, ports might use different file extensions for their documentation/ testing/ examples.
Instead, we should define what the `/share` folder should be used for, i.e. what files we expect there and disallow everything else. Furthermore, we should not (exclusively) rely on certain directories, and focus on the file extension we expect to be in `/share`. From my experience, this includes just `*.cmake`, `vcpkg_abi_info.txt`, `copyright` and `usage`.

**Reply by dg0yt** [reply]
> Instead, we should define what the `/share` folder should be used for, i.e. what files we expect there and disallow everything else. Furthermore, we should not (exclusively) rely on certain directories, and focus on the file extension we expect to be in `/share`. From my experience, this includes just `*.cmake`, `vcpkg_abi_info.txt`, `copyright` and `usage`.

It is not so simple. Some ports that I work with need data: gdal, proj. For my app, I may also need translations.

**Comment by JackBoosY** [comment]
This should be checked in the post check progress.

**Reply by dg0yt** [reply]
But it can only be a strict check when all ports are fixed...
